contracts/jobs/JobsManager.sol

```diff
@@ -70,94 +163,300 @@ contract JobsManager is IJobsManager, Verifiable, Controllable {
+
+        // Job must not already have an end block
+        require(job.endBlock == 0);
+        // Sender must be the job's broadcaster or elected transcoder
+        require(job.broadcasterAddress == msg.sender || jobs[_jobId].transcoderAddress == msg.sender);
+
+        // Set end block for job
+        job.endBlock = block.number.add(jobEndingPeriod);
+    }
+
+    /*
+     * @dev Submit claim for a range of segments
+     * @param _jobId Job identifier
+     * @param _segmentRange Range of claimed segments
+     * @param _claimRoot Merkle root of transcoded segment proof data for claimed segments
+     */
+    function claimWork(uint256 _jobId, uint256[2] _segmentRange, bytes32 _claimRoot) jobExists(_jobId) external returns (bool) {
+        Job storage job = jobs[_jobId];
+
+        // Job must be active
+        require(jobStatus(_jobId) == JobStatus.Active);
+        // Sender must be elected transcoder
+        require(job.transcoderAddress == msg.sender);
+
+        // Move fees from broadcaster deposit to escrow
+        uint256 fees = _segmentRange[1].sub(_segmentRange[0]).add(1).mul(job.maxPricePerSegment);
+        broadcasterDeposits[job.broadcasterAddress] = broadcasterDeposits[job.broadcasterAddress].sub(fees);
+        job.escrow = job.escrow.add(fees);
+
+        job.claims.push(Claim({
+            claimId: job.claims.length,
+            segmentRange: _segmentRange,
+            claimRoot: _claimRoot,
+            claimBlock: block.number,
+            transcoderTotalStake: bondingManager().transcoderTotalStake(msg.sender),
+            endVerificationBlock: block.number.add(verificationPeriod),
+            endSlashingBlock: block.number.add(verificationPeriod).add(slashingPeriod),
+            status: ClaimStatus.Pending
+        }));
 
-        // Check if there is an elected current active transcoder
-        if (electedTranscoder == address(0)) throw;
+        NewClaim(job.transcoderAddress, _jobId, job.claims.length - 1);
 
-        return jobs.newJob(_streamId, _transcodingOptions, _maxPricePerSegment, electedTranscoder);
+        return true;
     }
 
     /*
-     * @dev Return job details
+     * @dev Submit transcode receipt and invoke transcoding verification
      * @param _jobId Job identifier
+     * @param _segmentNumber Segment sequence number in stream
+     * @param _dataHash Content-addressed storage hash of segment data
+     * @param _transcodedDataHash Hash of transcoded segment data
+     * @param _broadcasterSig Broadcaster's signature over segment hash
+     * @param _proof Merkle proof for transcode receipt
      */
-    function getJobDetails(uint256 _jobId) public constant returns (uint256, uint256, address, address, uint256) {
-        return jobs.getJobDetails(_jobId);
+    function verify(
+        uint256 _jobId,
+        uint256 _claimId,
+        uint256 _segmentNumber,
+        string _dataHash,
+        string _transcodedDataHash,
+        bytes _broadcasterSig,
```

Comment:
I think `bytes` is an alias for `string` right? We can probably specify the length of these hashes/signatures.